### PR TITLE
Add a new AdvancedSample test app for messaging

### PR DESCRIPTION
**Pull request by chliangGoogle**
Create a new test app that is based on the Sample test app and move shared code into a shared folder.
Will add targets for app extensions in another PR.
#no-changelog

### CODE REVIEW

File: FirebaseMessaging/Apps/README.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+### Firebase Messaging Test Apps
+
+We offer two test apps: one in the Sample folder is a simple test app with the minimum Firebase Messaging code you need to enable delivering notifications. We kept the code minimum for you to easy debug and compare the sample code. The one inside the AdvancedSample folder is a test app based on the test app in the Sample folder. Actually they share most of the code in the main target. The AdvancedSample app provides advanced app extensions features, such as notification service extensions, watchkit, and app clips. So this app has more targets than the Sample app and requires your own provisioning profiles and certificates to be setup so that it can be run on a real device.
```
Comment (paulb777) [inline FirebaseMessaging/Apps/README.md:3]: Keep line limit to 100 so the README is readable in any editor

Comment (paulb777) [reply FirebaseMessaging/Apps/README.md:3]: s/minimum for you to easy/minimal for you to easily/

**Review by paulb777 — COMMENTED** [review]
Fix the CI and optionally migrate away from the deprecated IID calls.

File: FirebaseMessaging/Apps/README.md (hunk 1 above)
Comment (chliangGoogle) [reply FirebaseMessaging/Apps/README.md]: Done.

**Review by chliangGoogle — COMMENTED** [review]
We still need to keep IID code for debugging purpose when we remove InstanceID. We need to test cases when developers still install both IID and FCM when we removed the dependency.

**Review by paulb777 — APPROVED** [review]

File: FirebaseMessaging/Apps/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+### Firebase Messaging Test Apps
+
+We offer two test apps: one in the Sample folder is a simple test app with
+the minimum Firebase Messaging code you need to enable delivering
+notifications. We kept the code minimal for you to easily debug and compare
+the sample code. The one inside the AdvancedSample folder is a test app
+based on the test app in the Sample folder. Both Apps share most of the
+code in the main target. The AdvancedSample app provides advanced app
+extensions features, such as notification service extensions, watchkit, and
```
Comment (ryanwilson) [inline FirebaseMessaging/Apps/README.md:9]: s/watchkit/a watchOS app/

File: FirebaseMessaging/Apps/README.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+app clips. The advanced sample app has more targets than the Sample app and
```
Comment (ryanwilson) [inline FirebaseMessaging/Apps/README.md:10]: s/app clips/App Clips/

**Review by ryanwilson — APPROVED** [review]

File: FirebaseMessaging/Apps/README.md (hunk 2 above)
Comment (chliangGoogle) [reply FirebaseMessaging/Apps/README.md:9]: Done

File: FirebaseMessaging/Apps/README.md (hunk 3 above)
Comment (chliangGoogle) [reply FirebaseMessaging/Apps/README.md]: Done.

**Pull request merged**